--- /templates/CRM/common/civicrm.settings.php.template
+++ /templates/CRM/common/civicrm.settings.php.template
@@ -54,6 +54,60 @@
   define( 'CIVICRM_UF', '%%cms%%');
 }
 
+ /**
+ * Pantheon Systems:
+ *
+ * Repopulate needed variables based on the Pantheon environment if applicable.
+ * http://www.kalamuna.com/news/civicrm-pantheon
+ *
+ */
+if (!empty($_SERVER['PRESSFLOW_SETTINGS'])) {
+  $env = json_decode($_SERVER['PRESSFLOW_SETTINGS'], TRUE);
+  if (!empty($env['conf']['pantheon_binding'])) {
+    $pantheon_db = $env['databases']['default']['default'];
+    $pantheon_conf = $env['conf'];
+
+    //user name and password
+    $db_string = $pantheon_db['driver'] . '://' . $pantheon_db['username'] . ':' . $pantheon_db['password'] . '@';
+    //host
+    $db_string .= 'dbserver.' . $pantheon_conf['pantheon_environment'] . '.' . $pantheon_conf['pantheon_site_uuid'] . '.drush.in' . ':' . $pantheon_db['port'];
+    // database
+    $db_string .= '/' . $pantheon_db['database'] . '?new_link=true';
+
+    // define the database strings
+    define('CIVICRM_UF_DSN', $db_string);
+    define('CIVICRM_DSN', $db_string);
+
+    // define the file paths
+    global $civicrm_root;
+
+    $civicrm_root = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/%%modulePath%%';
+    define('CIVICRM_TEMPLATE_COMPILEDIR', '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/templates_c/');
+
+    // Use Drupal base url and path
+    global $base_url, $base_path;
+    define( 'CIVICRM_UF_BASEURL', $base_url . '/');
+    define( 'CIVICRM_SITE_KEY', '%%siteKey%%' );
+
+    if ( $pantheon_conf['pantheon_environment'] == 'dev' || $pantheon_conf['pantheon_environment'] == 'test' ){
+      /**
+      * This setting logs all emails to a file. Useful for debugging any mail (or civimail) issues.
+      * This will not send any email, so ensure this is commented out in production
+      */
+     define( 'CIVICRM_MAIL_LOG', '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/sites/default/files/civicrm/templates_c/mail.log' );
+    }
+
+    // Add this line only once above any settings overrides
+    global $civicrm_setting;
+    $civicrm_setting['Directory Preferences']['uploadDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/upload/';
+    $civicrm_setting['Directory Preferences']['customFileUploadDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/custom/';
+    $civicrm_setting['Directory Preferences']['imageUploadDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/persist/contribute/';
+    $civicrm_setting['Directory Preferences']['extensionsDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/sites/all/extensions/';
+    $civicrm_setting['URL Preferences']['extensionsURL'] = $base_url . '/sites/all/extensions';
+
+  }
+} else {
+
 /**
  * Content Management System (CMS) Datasource:
  *
@@ -61,6 +115,10 @@
  * Datasource (DSN) format:
  *      define( 'CIVICRM_UF_DSN', 'mysql://cms_db_username:cms_db_password@db_server/cms_database?new_link=true');
  */
+
+// PANTHEON USERS - These settings are overridden above when running on Pantheon.
+// These settings are ONLY included here to remain compatible with other hosts.
+
 if (!defined('CIVICRM_UF_DSN')) {
   define( 'CIVICRM_UF_DSN'           , 'mysql://%%CMSdbUser%%:%%CMSdbPass%%@%%CMSdbHost%%/%%CMSdbName%%?new_link=true');
 }
@@ -84,6 +142,10 @@
  *      define( 'CIVICRM_DSN'         , 'mysql://civicrm:YOUR_PASSWORD@localhost/civicrm?new_link=true');
  *
  */
+
+// PANTHEON USERS - These settings are overridden above when running on Pantheon.
+// These settings are ONLY included here to remain compatible with other hosts.
+
 if (!defined('CIVICRM_DSN')) {
   define( 'CIVICRM_DSN'          , 'mysql://%%dbUser%%:%%dbPass%%@%%dbHost%%/%%dbName%%?new_link=true');
 }
@@ -256,6 +318,8 @@
   define( 'CIVICRM_DOMAIN_ID', 1);
 }
 
+} // end Pantheon check
+
 /**
  * Settings to enable external caching using a Memcache server.  This is an
  * advanced feature, and you should read and understand the documentation
@@ -265,6 +329,22 @@
  *
  * @see http://civicrm.org/node/126
  */
+
+/**
+* Redis Caching
+* Using Redis Caching via an extention instead of database -- uncomment below to use.
+* based on https://github.com/ChrisChinchilla/com.schwartzmedia.rediscache
+*/
+
+if (!empty($pantheon_conf['redis_client_host'])) {
+  define( 'CIVICRM_DB_CACHE_CLASS', 'Redis' );
+  define( 'CIVICRM_DB_CACHE_HOST', $pantheon_conf['redis_client_host'] );
+  define( 'CIVICRM_DB_CACHE_PORT', $pantheon_conf['redis_client_port'] );
+  define( 'CIVICRM_DB_CACHE_PASSWORD', $pantheon_conf['redis_client_password'] );
+  define( 'CIVICRM_DB_CACHE_TIMEOUT', 3600 );
+  define( 'CIVICRM_DB_CACHE_PREFIX', 'pantheon-redis-civicrm:' );
+
+} else {
 
 /**
  * If you have a memcache server configured and want CiviCRM to make use of it,
@@ -317,6 +397,8 @@
 if (!defined('CIVICRM_MEMCACHE_PREFIX')) {
   define('CIVICRM_MEMCACHE_PREFIX', '');
 }
+
+} // end Pantheon Redis check
 
 /**
  * If you have multilingual site and you are using the "inherit CMS language"
