--- a/templates/CRM/common/civicrm.settings.php.template
+++ b/templates/CRM/common/civicrm.settings.php.template
@@ -63,6 +63,28 @@
 }
 
 /**
+* Pantheon Systems:
+*
+* Repopulate needed variables based on the Pantheon environment if applicable.
+* http://www.kalamuna.com/news/civicrm-pantheon
+*
+*/
+if (!empty($_SERVER['PRESSFLOW_SETTINGS'])) {
+  $env = json_decode($_SERVER['PRESSFLOW_SETTINGS'], TRUE);
+  if (!empty($env['conf']['pantheon_binding'])) {
+    $pantheon_db = $env['databases']['default']['default'];
+    $pantheon_conf = $env['conf'];
+
+    //user name and password
+    $pantheon_db_string = $pantheon_db['driver'] . '://' . $pantheon_db['username'] . ':' . $pantheon_db['password'] . '@';
+    //host
+    $pantheon_db_string .= 'dbserver.' . $pantheon_conf['pantheon_environment'] . '.' . $pantheon_conf['pantheon_site_uuid'] . '.drush.in' . ':' . $pantheon_db['port'];
+    // database
+    $pantheon_db_string .= '/' . $pantheon_db['database'] . '?new_link=true';
+  }
+}
+
+/**
  * Content Management System (CMS) Datasource:
  *
  * Update this setting with your CMS (Drupal, Backdrop CMS, or Joomla) database username, server and DB name.
@@ -70,7 +92,11 @@
  *      define( 'CIVICRM_UF_DSN', 'mysql://cms_db_username:cms_db_password@db_server/cms_database?new_link=true');
  */
 if (!defined('CIVICRM_UF_DSN') && CIVICRM_UF !== 'UnitTests') {
-  define( 'CIVICRM_UF_DSN'           , 'mysql://%%CMSdbUser%%:%%CMSdbPass%%@%%CMSdbHost%%/%%CMSdbName%%?new_link=true');
+  if (isset($pantheon_conf)) {
+    define('CIVICRM_UF_DSN', $pantheon_db_string);
+  } else {
+    define( 'CIVICRM_UF_DSN'           , 'mysql://%%CMSdbUser%%:%%CMSdbPass%%@%%CMSdbHost%%/%%CMSdbName%%?new_link=true');
+  }
 }
 
 /**
@@ -96,6 +122,9 @@
 if (!defined('CIVICRM_DSN')) {
   if (CIVICRM_UF === 'UnitTests' && isset($GLOBALS['_CV']['TEST_DB_DSN'])) {
     define('CIVICRM_DSN', $GLOBALS['_CV']['TEST_DB_DSN']);
+  } 
+  if (isset($pantheon_conf)) {
+    define('CIVICRM_DSN', $pantheon_db_string);
   }
   else {
     define('CIVICRM_DSN', 'mysql://%%dbUser%%:%%dbPass%%@%%dbHost%%/%%dbName%%?new_link=true');
@@ -162,9 +191,20 @@
 
 global $civicrm_root;
 
-$civicrm_root = '%%crmRoot%%';
+if (isset($pantheon_conf)) {
+  $civicrm_root = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/%%modulePath%%';
+}
+else {
+  $civicrm_root = '%%crmRoot%%';
+}
+
 if (!defined('CIVICRM_TEMPLATE_COMPILEDIR')) {
-  define( 'CIVICRM_TEMPLATE_COMPILEDIR', '%%templateCompileDir%%');
+  if (isset($pantheon_conf)) {
+    define('CIVICRM_TEMPLATE_COMPILEDIR', '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/templates_c/');
+  }
+  else {
+    define( 'CIVICRM_TEMPLATE_COMPILEDIR', '%%templateCompileDir%%');
+  }
 }
 
 /**
@@ -199,8 +239,10 @@
  *      define( 'CIVICRM_UF_BASEURL' , 'http://www.example.com/joomla/');
  *
  */
+global $base_url;
+ 
 if (!defined('CIVICRM_UF_BASEURL')) {
-  define( 'CIVICRM_UF_BASEURL'      , '%%baseURL%%');
+  define( 'CIVICRM_UF_BASEURL'      , $base_url);
 }
 
 /**
@@ -208,8 +250,25 @@
  *
  * Uncomment and edit the below as appropriate.
  */
- // Add this line only once above any settings overrides.
- //  global $civicrm_setting;
+
+global $civicrm_setting;
+ 
+if (isset($pantheon_conf)) {
+  $civicrm_setting['Directory Preferences']['uploadDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/upload/';
+  $civicrm_setting['Directory Preferences']['customFileUploadDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/custom/';
+  $civicrm_setting['Directory Preferences']['imageUploadDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/private/civicrm/persist/contribute/';
+  $civicrm_setting['Directory Preferences']['extensionsDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/sites/all/civicrm_custom/extensions/';
+  $civicrm_setting['URL Preferences']['extensionsURL'] = $base_url . '/sites/all/extensions';
+  
+  // Pantheon: Disable automatic download / installation of extensions on live
+  if ($pantheon_conf['pantheon_environment'] == 'live') {
+    $civicrm_setting['Extension Preferences']['ext_repo_url'] = false;
+  }
+  
+  // Pantheon: uncomment if customizing core PHP or template files
+  // $civicrm_setting['Directory Preferences']['customTemplateDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/sites/all/civicrm_custom/custom_templates/';
+  // $civicrm_setting['Directory Preferences']['customPHPPathDir'] = '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/sites/all/civicrm_custom/custom_php/';
+}
 
  // Override the Temporary Files directory.
  // $civicrm_setting['Directory Preferences']['customFileUploadDir'] = '/path/to/upload';
@@ -279,6 +338,12 @@
  * The CIVICRM_MAIL_LOG is a debug option which disables MTA (mail transport agent) interaction.
  * You must disable CIVICRM_MAIL_LOG before CiviCRM will talk to your MTA.
  */
+ 
+// Pantheon: if it's not the live site then send all emails to a log file
+if ($pantheon_conf['pantheon_environment'] != 'live') {
+  define('CIVICRM_MAIL_LOG', '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/code/sites/default/files/private/civicrm/ConfigAndLog/mail.log');
+}
+
 // if (!defined('CIVICRM_MAIL_LOG')) {
 // define( 'CIVICRM_MAIL_LOG', '%%templateCompileDir%%/mail.log');
 // }
@@ -312,54 +377,68 @@
  * To not use any caching (not recommended), use a value of 'NoCache'
  *
  */
-if (!defined('CIVICRM_DB_CACHE_CLASS')) {
-  define('CIVICRM_DB_CACHE_CLASS', 'ArrayCache');
-}
-
-/**
- * Change this to the IP address of your cache server if it is not on the
- * same machine (Unix).
- */
-if (!defined('CIVICRM_DB_CACHE_HOST')) {
-  define('CIVICRM_DB_CACHE_HOST', 'localhost');
-}
-
-/**
- * Change this if you are not using the standard port for your cache server.
- *
- * The standard port for Memcache & APCCache is 11211. For Redis it is 6379.
- */
-if (!defined('CIVICRM_DB_CACHE_PORT')) {
-  define('CIVICRM_DB_CACHE_PORT', 11211 );
-}
-
-/**
- * Change this if your cache server requires a password (currently only works
- * with Redis)
- */
-if (!defined('CIVICRM_DB_CACHE_PASSWORD')) {
-  define('CIVICRM_DB_CACHE_PASSWORD', '' );
-}
-
-/**
- * Items in cache will expire after the number of seconds specified here.
- * Default value is 3600 (i.e., after an hour)
- */
-if (!defined('CIVICRM_DB_CACHE_TIMEOUT')) {
-  define('CIVICRM_DB_CACHE_TIMEOUT', 3600 );
-}
-
-/**
- * If you are sharing the same cache instance with more than one CiviCRM
- * database, you will need to set a different value for the following argument
- * so that each copy of CiviCRM will not interfere with other copies.  If you only
- * have one copy of CiviCRM, you may leave this set to ''.  A good value for
- * this if you have two servers might be 'server1_' for the first server, and
- * 'server2_' for the second server.
- */
-if (!defined('CIVICRM_DB_CACHE_PREFIX')) {
-  define('CIVICRM_DB_CACHE_PREFIX', '');
-}
+ 
+if (isset($pantheon_conf['redis_client_host']) && !empty($pantheon_conf['redis_client_host'])) {
+  define('CIVICRM_DB_CACHE_CLASS', 'Redis');
+  define('CIVICRM_DB_CACHE_HOST', $pantheon_conf['redis_client_host']);
+  define('CIVICRM_DB_CACHE_PORT', $pantheon_conf['redis_client_port']);
+  define('CIVICRM_DB_CACHE_PASSWORD', $pantheon_conf['redis_client_password']);
+  define('CIVICRM_DB_CACHE_TIMEOUT', 3600);
+  define('CIVICRM_DB_CACHE_PREFIX', 'pantheon-redis-civicrm:');
+  
+} 
+else {
+
+  if (!defined('CIVICRM_DB_CACHE_CLASS')) {
+    define('CIVICRM_DB_CACHE_CLASS', 'ArrayCache');
+  }
+
+  /**
+  * Change this to the IP address of your cache server if it is not on the
+  * same machine (Unix).
+  */
+  if (!defined('CIVICRM_DB_CACHE_HOST')) {
+    define('CIVICRM_DB_CACHE_HOST', 'localhost');
+  }
+
+  /**
+  * Change this if you are not using the standard port for your cache server.
+  *
+  * The standard port for Memcache & APCCache is 11211. For Redis it is 6379.
+  */
+  if (!defined('CIVICRM_DB_CACHE_PORT')) {
+    define('CIVICRM_DB_CACHE_PORT', 11211 );
+  }
+
+  /**
+  * Change this if your cache server requires a password (currently only works
+  * with Redis)
+  */
+  if (!defined('CIVICRM_DB_CACHE_PASSWORD')) {
+    define('CIVICRM_DB_CACHE_PASSWORD', '' );
+  }
+
+  /**
+  * Items in cache will expire after the number of seconds specified here.
+  * Default value is 3600 (i.e., after an hour)
+  */
+  if (!defined('CIVICRM_DB_CACHE_TIMEOUT')) {
+    define('CIVICRM_DB_CACHE_TIMEOUT', 3600 );
+  }
+
+  /**
+  * If you are sharing the same cache instance with more than one CiviCRM
+  * database, you will need to set a different value for the following argument
+  * so that each copy of CiviCRM will not interfere with other copies.  If you only
+  * have one copy of CiviCRM, you may leave this set to ''.  A good value for
+  * this if you have two servers might be 'server1_' for the first server, and
+  * 'server2_' for the second server.
+  */
+  if (!defined('CIVICRM_DB_CACHE_PREFIX')) {
+    define('CIVICRM_DB_CACHE_PREFIX', '');
+  }
+
+} // end Pantheon Redis check
 
 /**
  * If you have multilingual site and you are using the "inherit CMS language"
