diff --git a/drupal/civicrm.install b/drupal/civicrm.install
index 7e38031fd..5bfa1e690 100644
--- a/drupal/civicrm.install
+++ b/drupal/civicrm.install
@@ -92,7 +92,9 @@ function civicrm_requirements($phase) {
     $civicrm_path = substr_replace($civicrm_path, '', $pos, strlen($civicrm_path));
   }
 
-  $url = $base_url . '/' . $civicrm_path . 'install/index.php';
+  // Pass variables already configured
+  $querystring = isset($_GET['profile']) ? '?profile=' . $_GET['profile'] . '&locale=' . $_GET['locale'] : NULL;
+  $url = $base_url . '/' . $civicrm_path . 'install/index.php' . $querystring;
 
   $settings = glob('sites/*/civicrm.settings.php');
   $problems = array();
@@ -101,7 +103,7 @@ function civicrm_requirements($phase) {
     $problems[] = array(
       'title' => $t('CiviCRM settings does not exist'),
       'value' =>
-      $t('CiviCRM settings file does not exist. It should be created by CiviCRM <a href="!link">install</a>',
+      t('CiviCRM has not been installed. The CiviCRM Starter Kit requires CiviCRM to be installed before continuing. <a href="!link">Please install CiviCRM now</a>.',
         array('!link' => $url)),
       'severity' => REQUIREMENT_ERROR,
     );
