diff --git a/CRM/Utils/File.php b/CRM/Utils/File.php
index c69d9c8..7c5f7c5 100644
--- a/CRM/Utils/File.php
+++ b/CRM/Utils/File.php
@@ -566,13 +566,32 @@ HTACCESS;
   }

   /**
-   * Create the base file path from which all our internal directories are
-   * offset. This is derived from the template compile directory set
+   * Create the base public file path from which all our public internal
+   * directories are offset. CIVICRM_PUBLIC_FILES_DIR is set in civicrm.settings.php.
    */
   public static function baseFilePath() {
     static $_path = NULL;
     if (!$_path) {
       // Note: Don't rely on $config; that creates a dependency loop.
+      if (!defined('CIVICRM_PUBLIC_FILES_DIR')) {
+        throw new RuntimeException("Undefined constant: CIVICRM_PUBLIC_FILES_DIR");
+      }
+
+      $path = CIVICRM_PUBLIC_FILES_DIR;
+
+      $_path = CRM_Utils_File::addTrailingSlash($path);
+    }
+    return $_path;
+  }
+
+  /**
+   * Create the base private file path from which all our internal directories are
+   * offset. This is derived from the template compile directory set
+   */
+  public static function basePrivateFilePath() {
+    static $_path = NULL;
+    if (!$_path) {
+      // Note: Don't rely on $config; that creates a dependency loop.
       if (!defined('CIVICRM_TEMPLATE_COMPILEDIR')) {
         throw new RuntimeException("Undefined constant: CIVICRM_TEMPLATE_COMPILEDIR");
       }
diff --git a/install/civicrm.php b/install/civicrm.php
index 50e4271..8a7c356 100644
--- a/install/civicrm.php
+++ b/install/civicrm.php
@@ -220,6 +220,7 @@ function civicrm_config(&$config) {
   $params = array(
     'crmRoot' => $crmPath,
     'templateCompileDir' => $compileDir,
+    'publicFilesDir' => $compileDir,
     'frontEnd' => 0,
     'dbUser' => addslashes($config['mysql']['username']),
     'dbPass' => addslashes($config['mysql']['password']),
diff --git a/templates/CRM/common/civicrm.settings.php.template b/templates/CRM/common/civicrm.settings.php.template
index 8582e59..fa897d5 100644
--- a/templates/CRM/common/civicrm.settings.php.template
+++ b/templates/CRM/common/civicrm.settings.php.template
@@ -162,6 +162,9 @@ if (!defined('CIVICRM_LOGGING_DSN')) {
  * IMPORTANT: The COMPILEDIR directory must exist,
  * and your web server must have read/write access to these directories.
  *
+ * IMPORTANT FOR NGINX SITES: You may need to define a file system path that is private,
+ * especially if you don't have access to NGINX configuration but you do have a path,
+ * for example, like: sites/default/files/private
  *
  * EXAMPLE - Drupal:
  * If the path to the Drupal home directory is /var/www/htdocs/drupal
@@ -216,6 +219,19 @@ if (!defined('CIVICRM_TEMPLATE_COMPILEDIR')) {
 }

 /**
+ * Provide a default public CiviCRM file path on Pantheon.
+ * For files that need to be publicly accessible such as the CKEditor configuration file.
+ */
+if (!defined('CIVICRM_PUBLIC_FILES_DIR')) {
+  if (isset($pantheon_conf)) {
+    define('CIVICRM_PUBLIC_FILES_DIR', '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/civicrm/');
+  }
+  else {
+    define( 'CIVICRM_PUBLIC_FILES_DIR', '%%publicFilesDir%%');
+  }
+}
+
+/**
  * Site URLs:
  *
  * This section defines absolute and relative URLs to access the host CMS (Backdrop, Drupal, or Joomla) resources.
