diff --git a/CRM/Utils/File.php b/CRM/Utils/File.php
index c69d9c8..2382f18 100644
--- a/CRM/Utils/File.php
+++ b/CRM/Utils/File.php
@@ -595,6 +595,25 @@ HTACCESS;
   }

   /**
+   * Create the base public file path from which all our public internal
+   * directories are offset. This is set in civicrm.settings.php.
+   */
+  public static function basePublicFilePath() {
+    static $_path = NULL;
+    if (!$_path) {
+      // Note: Don't rely on $config; that creates a dependency loop.
+      if (!defined('CIVICRM_PUBLIC_FILES_DIR')) {
+        throw new RuntimeException("Undefined constant: CIVICRM_PUBLIC_FILES_DIR");
+      }
+
+      $path = CIVICRM_PUBLIC_FILES_DIR;
+
+      $_path = CRM_Utils_File::addTrailingSlash($path);
+    }
+    return $_path;
+  }
+
+  /**
    * Determine if a path is absolute.
    *
    * @param string $path
@@ -630,8 +649,8 @@ HTACCESS;
       return $directory;
     }

-    // make everything relative from the baseFilePath
-    $basePath = self::baseFilePath();
+    // make everything relative from the basePublicFilePath
+    $basePath = self::basePublicFilePath();
     // check if basePath is a substr of $directory, if so
     // return rest of string
     if (substr($directory, 0, strlen($basePath)) == $basePath) {
diff --git a/CRM/Utils/System/Base.php b/CRM/Utils/System/Base.php
index a66fba1..5ccfe49 100644
--- a/CRM/Utils/System/Base.php
+++ b/CRM/Utils/System/Base.php
@@ -611,7 +611,7 @@ abstract class CRM_Utils_System_Base {

     return array(
       'url' => $filesURL,
-      'path' => CRM_Utils_File::baseFilePath(),
+      'path' => CRM_Utils_File::basePublicFilePath(),
     );
   }

diff --git a/CRM/Utils/System/WordPress.php b/CRM/Utils/System/WordPress.php
index d49c7d7..9ba22b1 100644
--- a/CRM/Utils/System/WordPress.php
+++ b/CRM/Utils/System/WordPress.php
@@ -79,7 +79,7 @@ class CRM_Utils_System_WordPress extends CRM_Utils_System_Base {
     $config = CRM_Core_Config::singleton();
     $cmsUrl = CRM_Utils_System::languageNegotiationURL($config->userFrameworkBaseURL, FALSE, TRUE);
     $cmsPath = $this->cmsRootPath();
-    $filesPath = CRM_Utils_File::baseFilePath();
+    $filesPath = CRM_Utils_File::basePublicFilePath();
     $filesRelPath = CRM_Utils_File::relativize($filesPath, $cmsPath);
     $filesURL = rtrim($cmsUrl, '/') . '/' . ltrim($filesRelPath, ' /');
     return array(
diff --git a/install/civicrm.php b/install/civicrm.php
index 945b492..1b9a67f 100644
--- a/install/civicrm.php
+++ b/install/civicrm.php
@@ -220,20 +220,21 @@ function civicrm_config(&$config) {
   $params = array(
     'crmRoot' => $crmPath,
     'templateCompileDir' => $compileDir,
+    'publicFilesDir' => $compileDir,
     'frontEnd' => 0,
     'dbUser' => addslashes($config['mysql']['username']),
     'dbPass' => addslashes($config['mysql']['password']),
     'dbHost' => $config['mysql']['server'],
     'dbName' => addslashes($config['mysql']['database']),
   );
-
+
   // when running on Pantheon, part of $crmPath is set dynamically in civicrm_settings.php
   //if civicrm is in a profile
   if (strpos($crmPath , 'profile')) {
     $modulePathParts = explode('profiles/', $crmPath);
     $params['modulePath'] = 'profiles/' . $modulePathParts[1];
   }
-  // if civicrm is not in profile, it is in sites
+  // if civicrm is not in profile, it is in sites
   if (!isset($params['modulePath'])) {
     $modulePathParts = explode('sites/', $crmPath);
     $params['modulePath'] = 'sites/' . $modulePathParts[1];
diff --git a/templates/CRM/common/civicrm.settings.php.template b/templates/CRM/common/civicrm.settings.php.template
index 35c83d8..ad56947 100644
--- a/templates/CRM/common/civicrm.settings.php.template
+++ b/templates/CRM/common/civicrm.settings.php.template
@@ -162,6 +162,9 @@ if (!defined('CIVICRM_LOGGING_DSN')) {
  * IMPORTANT: The COMPILEDIR directory must exist,
  * and your web server must have read/write access to these directories.
  *
+ * IMPORTANT FOR NGINX SITES: You may need to define a file system path that is private,
+ * especially if you don't have access to NGINX configuration but you do have a path,
+ * for example, like: sites/default/files/private
  *
  * EXAMPLE - Drupal:
  * If the path to the Drupal home directory is /var/www/htdocs/drupal
@@ -216,6 +219,19 @@ if (!defined('CIVICRM_TEMPLATE_COMPILEDIR')) {
 }

 /**
+ * Provide a default public CiviCRM file path on Pantheon.
+ * For files that need to be publicly accessible such as the CKEditor configuration file.
+ */
+if (!defined('CIVICRM_PUBLIC_FILES_DIR')) {
+  if (isset($pantheon_conf)) {
+    define('CIVICRM_PUBLIC_FILES_DIR', '/srv/bindings/' . $pantheon_conf['pantheon_binding'] . '/files/civicrm/');
+  }
+  else {
+    define( 'CIVICRM_PUBLIC_FILES_DIR', '%%publicFilesDir%%');
+  }
+}
+
+/**
  * Site URLs:
  *
  * This section defines absolute and relative URLs to access the host CMS (Backdrop, Drupal, or Joomla) resources.
