diff --git a/CRM/Core/Config.php b/CRM/Core/Config.php
index 49ae502..70b3cd4 100644
--- a/CRM/Core/Config.php
+++ b/CRM/Core/Config.php
@@ -183,8 +183,9 @@ class CRM_Core_Config extends CRM_Core_Config_Variables {
       }
 
       // first, attempt to get configuration object from cache
-      $cache = CRM_Utils_Cache::singleton();
-      self::$_singleton = $cache->get('CRM_Core_Config' . CRM_Core_Config::domainID());
+      // no, not going to skip cache if we're on Pantheon
+      //$cache = CRM_Utils_Cache::singleton();
+      //self::$_singleton = $cache->get('CRM_Core_Config' . CRM_Core_Config::domainID());
       // if not in cache, fire off config construction
       if (!self::$_singleton) {
         self::$_singleton = new CRM_Core_Config;
@@ -204,7 +205,8 @@ class CRM_Core_Config extends CRM_Core_Config_Variables {
           // We can safely delete this once we release 4.4.4
           // self::$_singleton->setCoreVariables();
         }
-        $cache->set('CRM_Core_Config' . CRM_Core_Config::domainID(), self::$_singleton);
+        // We don't care about caching config on Pantheon
+        //$cache->set('CRM_Core_Config' . CRM_Core_Config::domainID(), self::$_singleton);
       }
       else {
         // we retrieve the object from memcache, so we now initialize the objects
