diff --git a/extern/ipn.php b/extern/ipn.php
index a7f56d1..6f479af 100644
--- a/extern/ipn.php
+++ b/extern/ipn.php
@@ -41,6 +41,9 @@ require_once '../civicrm.config.php';
 /* Cache the real UF, override it with the SOAP environment */
 
 $config = CRM_Core_Config::singleton();
+
+CRM_Utils_System::loadBootStrap(array(), FALSE);
+
 $log = new CRM_Utils_SystemLogger();
 if (empty($_GET)) {
   $log->alert('payment_notification processor_name=PayPal', $_REQUEST);
