diff --git a/extern/ipn.php b/extern/ipn.php
index c3ded9a..0262e6a 100644
--- a/extern/ipn.php
+++ b/extern/ipn.php
@@ -42,15 +42,8 @@ require_once '../civicrm.config.php';

 $config = CRM_Core_Config::singleton();
 $log = new CRM_Utils_SystemLogger();
-if (empty($_GET)) {
-  $log->alert('payment_notification processor_name=PayPal', $_REQUEST);
-  $paypalIPN = new CRM_Core_Payment_PayPalProIPN($_REQUEST);
-}
-else {
-  $log->alert('payment_notification PayPal_Standard', $_REQUEST);
-  $paypalIPN = new CRM_Core_Payment_PayPalIPN($_REQUEST);
-  // @todo upgrade standard per Pro
-}
+$log->alert('payment_notification processor_name=PayPal', $_REQUEST);
+$paypalIPN = new CRM_Core_Payment_PayPalProIPN($_REQUEST);
 try {
   //CRM-18245
   if ($config->userFramework == 'Joomla') {
