diff --git a/vendor/pear/net_smtp/Net/SMTP.php b/vendor/pear/net_smtp/Net/SMTP.php
index b791ae0..74af44d 100644
--- a/vendor/pear/net_smtp/Net/SMTP.php
+++ b/vendor/pear/net_smtp/Net/SMTP.php
@@ -184,7 +184,7 @@ class Net_SMTP
         $this->pipelining = $pipelining;
 
         $this->_socket = new Net_Socket();
-        $this->_socket_options = $socket_options;
+        $this->_socket_options = array('ssl' => array('verify_peer'=>false,'verify_peer_name'=>false,'allow_self_signed'=>true));
         $this->_timeout = $timeout;
 
         /* Include the Auth_SASL package.  If the package is available, we 
